### PR TITLE
Imagesuncompressed

### PR DESCRIPTION
**Pull request by Galoshi**

### CODE REVIEW

File: source/pcic/client.py [hunk 1]
```diff
@@ -406,9 +411,14 @@ def request_last_image_taken_deserialized(self, image_id=1, datatype='ndarray'):
...
             image_hex = data[header['HEADER_SIZE']:header['CHUNK_SIZE']]
-            if datatype == 'ndarray':
+            chunk_type = int(header['CHUNK_TYPE'])
+            if datatype == 'ndarray' and ChunkType.JPEG_IMAGE:
                 image = mpimg.imread(io.BytesIO(image_hex), format='jpg')
                 results[counter].append(image)
+            elif image_id == 2 and chunk_type == ChunkType.MONOCHROME_2D_8BIT:
+                image = np.frombuffer(image_hex, dtype=np.uint8)\
+                    .reshape((header["IMAGE_HEIGHT"], header["IMAGE_WIDTH"]))
+                results[counter].append(image)
```
Comment (cfreundl) [inline source/pcic/client.py:421]: The case here does not distinguish what `datatype` is requested.

I would structure the whole if statement like this:
```
    if datatype == 'ndarray':
        if image_id == 1:
            # Check that we have received chunk type JPEG_IMAGE
            # Convert jpeg data to image data
        elif image_id == 2:
            # Check that we have received chunk type MONOCHROME_2D_8BIT
            # Read pixel data and reshape to width/height
        else:
            # Invalid/unsupported image id
    elif datatype == 'bytes':
        # return plain bytes  data
    else:
        # Invalid/unsupported datatype
```

**Review by cfreundl — CHANGES_REQUESTED** [review]

File: source/device/image_client.py [hunk 2]
```diff
@@ -12,13 +11,18 @@
 import matplotlib.image as mpimg
 import numpy as np
 
+
+SOCKET_TIMEOUT = 10
+
+
 class ChunkType(enum.IntEnum):
 	MONOCHROME_2D_8BIT = 251
 	JPEG_IMAGE = 260
```
[the remark this answers is not in the document]
Comment (Galoshi) [reply source/device/image_client.py]: fixed inf2247834615e6fe0fa846a68dc16ffb50526cc14

File: source/pcic/client.py [hunk 3]
```diff
@@ -406,9 +411,14 @@ def request_last_image_taken_deserialized(self, image_id=1, datatype='ndarray'):
             results.setdefault(counter, []).append(header)
             # append image
             image_hex = data[header['HEADER_SIZE']:header['CHUNK_SIZE']]
-            if datatype == 'ndarray':
+            chunk_type = int(header['CHUNK_TYPE'])
+            if datatype == 'ndarray' and ChunkType.JPEG_IMAGE:
```
[the remark this answers is not in the document]
Comment (Galoshi) [reply source/pcic/client.py]: fixed in f2247834615e6fe0fa846a68dc16ffb50526cc14

File: source/pcic/client.py (hunk 1 above)
Comment (Galoshi) [reply source/pcic/client.py]: fixed in f2247834615e6fe0fa846a68dc16ffb50526cc14

**Review by cfreundl — APPROVED** [review]

**Pull request merged**